### ISSUE TITLE
Docs: Example of tracer.captureMethod() should demo in async instead of sync function

### ISSUE THREAD

**Issue by kennethwkz-mm** [aws-powertools/powertools-lambda-typescript#1203]
### What were you searching in the docs?

About the tracer.captureMethod() example of correct usage, actually the method is not support non-promise function.

### Is this related to an existing documentation section?

https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/tracer/#methods

### How can we improve?

```Markdown
It should change the example with show @tracer.captureMethod() in async - promise function.
```

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

**Comment by dreamorosi** [comment]
Thank you @kennethwkz-mm for spotting and reporting this. We do indeed need to make the change as it could be confusing for folks copy/pasting the example (as proven by our conversation on Discord).

I have added the `help-wanted` & `good-first-issue` labels. If anyone wants to pick this up, please leave a comment below to discuss and have the issue assigned to you before opening a PR. Apart from that, it should be a fairly straightforward change so I'll be happy to review & merge it once it's up.

**Comment by dreamorosi** [comment]
As part of #1219 we are reworking some of the code snippets of that page, we will address this issue as part of that work.

**Comment by niko-achilles** [comment]
this is addressed in PR #1229
